options/cf_options.cc

```diff
@@ -451,184 +549,224 @@ std::unordered_map<std::string, OptionTypeInfo>
           }}},
         {"table_factory",
          {offset_of(&ColumnFamilyOptions::table_factory),
-          OptionType::kTableFactory, OptionVerificationType::kByName,
-          OptionTypeFlags::kCompareLoose, 0}},
+          OptionType::kConfigurable, OptionVerificationType::kByName,
+          (OptionTypeFlags::kShared | OptionTypeFlags::kCompareLoose |
```

Comment:
We may need to give the guidelines for developer if they are adding new entry like table_tactoy or other factory, how to decide the OptionTypeFlags. For example, tt might be hard to understand why it is "OptionTypeFlags::kShared | OptionTypeFlags::kCompareLoose | OptionTypeFlags::kDontPrepare".

Reply:
Added more comments